sql-plugin/src/main/scala/org/apache/spark/sql/rapids/misc.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.rapids
+
+import ai.rapids.cudf.{ColumnVector}
+import com.nvidia.spark.rapids.{Arm, GpuColumnVector, GpuUnaryExpression}
+
+import org.apache.spark.sql.catalyst.expressions.{ExpectsInputTypes, Expression}
+import org.apache.spark.sql.types.{AbstractDataType, DataType, NullType, StringType}
+
+case class GpuRaiseError(child: Expression) extends GpuUnaryExpression with ExpectsInputTypes
+  with Arm {
+
+  override def dataType: DataType = NullType
+  override def inputTypes: Seq[AbstractDataType] = Seq(StringType)
+  override def toString: String = s"raise_error($child)"
+
+  /** Could evaluating this expression cause side-effects, such as throwing an exception? */
+  override def hasSideEffects: Boolean = true
+
+  override protected def doColumnar(input: GpuColumnVector): ColumnVector = {
+    if (input.getRowCount <= 0) {
+      // For the case: when(condition, raise_error())
```

Comment:
it would be nice to cover this specific case in the python tests, it doesn't seem like we are.

Reply:
Actually, according to spark, I don't think this `raise_error()` (no args) is possible:
```
pyspark.sql.utils.AnalysisException: Invalid number of arguments for function raise_error. Expected: 1; Found: 0; line 1 pos 7
```

Reply:
This is possible and should be tested, e.g.:
```
>>> import pyspark.sql.functions as f
>>> df = spark.range(0)
>>> df.count()
0
>>> df.select(f.raise_error(f.col("id"))).explain()
== Physical Plan ==
*(1) Project [raise_error(cast(id#12L as string), NullType) AS raise_error(id)#20]
+- *(1) Range (0, 0, step=1, splits=12)


>>> df.select(f.raise_error(f.col("id"))).collect()
[]
```

Reply:
Yeah, raise_error needs to accept the parameter and I just updated the comment.


> Actually, according to spark, I don't think this `raise_error()` (no args) is possible:
> 
> ```
> pyspark.sql.utils.AnalysisException: Invalid number of arguments for function raise_error. Expected: 1; Found: 0; line 1 pos 7
> ```

Yeah, raise_error needs to accept the parameter and I just updated the comment.

Reply:
> This is possible and should be tested, e.g.:
> 
> ```
> >>> import pyspark.sql.functions as f
> >>> df = spark.range(0)
> >>> df.count()
> 0
> >>> df.select(f.raise_error(f.col("id"))).explain()
> == Physical Plan ==
> *(1) Project [raise_error(cast(id#12L as string), NullType) AS raise_error(id)#20]
> +- *(1) Range (0, 0, step=1, splits=12)
> 
> 
> >>> df.select(f.raise_error(f.col("id"))).collect()
> []
> ```

Done